src/app/applyContextMenu.js

```diff
@@ -0,0 +1,112 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { clipboard, Menu, MenuItem } from 'electron'
+import { isExternalLink } from './externalLinkHandlers.js'
+
+/**
+ * Enable context menu by right click
+ *
+ * @param {import('electron').BrowserWindow} browserWindow - Browser window
+ */
+export function applyContextMenu(browserWindow) {
+	browserWindow.webContents.on('context-menu', (event, params) => {
+		const menu = new Menu()
+
+		// Add context actions for misspelling words and typos
+		const menuMisspellingItems = [
+			...params.dictionarySuggestions.map(suggestion => ({
+				label: suggestion,
+				click: () => browserWindow.webContents.replaceMisspelling(suggestion),
+			})),
+			{ type: 'separator' },
+			{
+				label: 'Add to dictionary',
+				click: () => browserWindow.webContents.session.addWordToSpellCheckerDictionary(params.misspelledWord),
+			},
+			{ type: 'separator' },
+		]
+		if (params.misspelledWord) {
+			menuMisspellingItems.forEach(item => menu.append(new MenuItem(item)))
+		}
+
+		// Add context actions for handling images
+		const menuImageItems = [
+			{
+				label: 'Copy image',
+				click: () => browserWindow.webContents.copyImageAt(params.x, params.y),
+			},
+			{
+				label: 'Save image',
+				click: () => browserWindow.webContents.downloadURL(params.srcURL),
+			},
+			{ type: 'separator' },
+		]
+		if (params.hasImageContents) {
+			menuImageItems.forEach(item => menu.append(new MenuItem(item)))
+		}
+
+		// Add context actions for handling links
+		const menuLinkItems = [
+			{
+				label: 'Copy link address',
+				click: () => clipboard.writeText(params.linkURL),
+			},
+			{
+				label: 'Copy link text',
+				click: () => clipboard.writeText(params.linkText.trim() || params.linkURL),
+			},
+			{ type: 'separator' },
+		]
+		if (params.linkURL && isExternalLink(params.linkURL)) {
+			menuLinkItems.forEach(item => menu.append(new MenuItem(item)))
+		}
+
+		// Add context actions for clipboard events and text editing
+		const menuClipboardItems = [
+			{
+				role: 'copy',
+				enabled: params.selectionText && params.editFlags.canCopy,
+			},
+			{
+				role: 'cut',
+				enabled: params.selectionText && params.isEditable && params.editFlags.canCut,
+			},
+			{
+				role: 'selectAll',
+				enabled: params.editFlags.canSelectAll,
+			},
+			{
+				role: 'paste',
+				enabled: params.isEditable && params.editFlags.canPaste,
+			},
+			{ type: 'separator' },
+		]
+		if (params.isEditable || params.selectionText.length) {
+			menuClipboardItems.forEach(item => menu.append(new MenuItem(item)))
+		}
```

Comment:
Editable context and text context are different contexts of the UI.

Editable context is the context with something that can be edited. In this context, a user has edit content abilities (copy/past), even if some of them could not be available at a specific moment (e.g. "cut" is disabled when there is nothing to cut).

"Selectable" context is a context with something selectable. In this context, a user can select and copy selected (if there is anything to copy, anything selected). But there are no editing operations (cut/paste) by design in this context.

Different **contexts** should have different **context menus**. With disabling for unavailable operations of the current context.

This behavior changed with the last comment:

![image](https://user-images.githubusercontent.com/25978914/230063106-d8b526ed-ae0a-438f-aa44-760d86023421.png)